apiary.apib

```diff
@@ -73,17 +73,47 @@ If you find any bug in our API please use [admin@zonky.cz](mailto:admin@zonky.cz
 
 # OAuth 2 authentication
 
-Zonky users are identified using [OAuth 2.0](http://oauth.net/2/) protocol. At the moment we support only OAuth 2 
+Zonky users are identified using [OAuth 2.0](http://oauth.net/2/) protocol. At the moment we support OAuth 2
 [Resource Owner Password Credentials grant flow](https://tools.ietf.org/html/rfc6749#section-4.3) where user receives an access token and
-a refresh token in exchange for their username and password. 
+a refresh token in exchange for their username and password and [Authorization Code Grant Flow ](https://tools.ietf.org/html/rfc6749#page-24) where
+client app obtains authorization code which is later exchanged for access token
 
 **Acting** on **behalf** of other **users** by supplying their **credentials** is strictly **prohibited**. Applications violating this rule will be banned with no mercy. We may eventually extend our implemenation with [Authorization Code Grant flow](https://tools.ietf.org/html/rfc6749#section-4.1) allowing  3rd party applications to act on behalf of a registered user.
 
 ## Access token and refresh token
-To keep things as secure as possible you have to pay attention to the right usage of OAuth authentication protocol. The right way is to obtain 
+To keep things as secure as possible you have to pay attention to the right usage of OAuth authentication protocol. The right way is to obtain
 an access token and a refresh token with user's credentials *on the first authentication request*. Retrieved access token has limited validity (advertised in the
 response). After its expiration you should retrieve a new one with a refresh token which has longer validity.
 
+## Authorization code grant Flow
+### Client onboarding
+Client application must obtain client credentials before it can use authorization code flow. To start onboarding process please send us email to oauth@zonky.cz
+with following details
+- application name
+- application description
+- requested scopes (list of scopes this app will need to operate)
+- redirect url ( target url where is user redirected after granting authorization code or in case of error) Url is verified for match except for params
+
+Zonky will evaluate your request and if everything is fine we will provide unique clientId and clientSecret for you. **ClientSecret is confidential** and should not
```

Comment:
How do you expect open source applications to handle this requirement? Or are open-source clients, such as _RoboZonky_, expected to keep using the old mechanism?

Reply:
If I understand it correctly every user of roboZonky is using its own instance. There is no big issue with using username and password as they are not providing it to any third party application. Authorization code is target at apps with own backend that are used by multiple users.

Reply:
Sounds good. If I can just keep doing what I've been doing until now, I'll be fine.

(And yes - people download and install _RoboZonky_ on their own. There is no middle-man, there is no _RoboZonky_ server that would relay requests. All communication goes directly from the user to the Zonky server.)